src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/LaunchProfiles/LaunchSettingsQueryVersionProviderExport.cs

```diff
@@ -0,0 +1,218 @@
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Query
+{
+    /// <summary>
+    /// Implementation of <see cref="IQueryDataSourceVersionProvider"/> for launch
+    /// settings. Receives data from the <see cref="LaunchSettingsQueryVersionProvider"/>
+    /// and passes it on to subscribed observers.
+    /// </summary>
+    /// <remarks>
+    /// We need to be careful that this type is only loaded by the Project Query API.
+    /// If our code were to load it, then we would cause the load of the Project Query
+    /// assemblies in situations where they weren't actually loaded. To avoid this, our
+    /// code has no direct dependency on this type, only on the <see cref="ILaunchSettingsVersionPublisher"/>
+    /// interface (which is defined in the project system). When this type *is* loaded it
+    /// registers itself with the <see cref="LaunchSettingsQueryVersionProvider"/>.
+    /// </remarks>
+    [Export(typeof(IQueryDataSourceVersionProvider))]
+    [ExportMetadata("Name", LaunchSettingsQueryVersionGroupName)]
+    internal class LaunchSettingsQueryVersionProviderExport : OnceInitializedOnceDisposed, IQueryDataSourceVersionProvider, ILaunchSettingsVersionPublisher
+    {
+        public const string LaunchSettingsQueryVersionGroupName = "LaunchSettings";
+
+        /// <remarks>
+        /// Responsible for generating new version information.
+        /// </remarks>
+        private readonly LaunchSettingsQueryVersionProvider _provider;
+        /// <remarks>
+        /// Processes notifications to the <see cref="_observers"/>.
+        /// </remarks>
+        private readonly ITargetBlock<Action> _processingBlock;
+        /// <summary>
+        /// The set of change notification observers.
+        /// </summary>
+        private ImmutableHashSet<IObserver<QueryDataSourceChangeNotification>> _observers = ImmutableHashSet<IObserver<QueryDataSourceChangeNotification>>.Empty;
+        /// <remarks>
+        /// Used to keep track of whether or not a new subscriber has received its initial snapshot.
+        /// </remarks>
+        private long _processedNotificationCount;
+        /// <remarks>
+        /// The current set of versions.
+        /// </remarks>
+        private QueryDataVersions _versions = QueryDataVersions.Empty;
+        /// <remarks>
+        /// The most recent set of versions sent to subscribers.
+        /// </remarks>
+        private QueryDataVersions _latestPostedVersions = QueryDataVersions.Empty;
+
+        [ImportingConstructor]
+        public LaunchSettingsQueryVersionProviderExport(LaunchSettingsQueryVersionProvider provider)
+        {
+            _provider = provider;
+#pragma warning disable RS0030 // Do not used banned APIs
+            _processingBlock = DataflowBlockSlim.CreateActionBlock<Action>(callback => callback(), nameFormat: nameof(LaunchSettingsQueryVersionProviderExport));
+#pragma warning restore RS0030 // Do not used banned APIs
```

Comment:
Why is this API banned?

Reply:
One of the BannedSymbols.txt files has the answer:
> `Use DataflowBlockFactory.CreateActionBlock to ensure failed blocks are reported as non-fatal watsons and assigned to the correct component`

`DataflowBlockFactory.CreateActionBlock` creates an action block with additional machinery for catching and reporting failures, and possibly recovering. In almost all cases this is preferred over what I am doing here.

However, `DataflowBlockFactory.CreateActionBlock` takes an `UnconfiguredProject` as one of its parameters, but this type (`LaunchSettingsQueryVersionProviderExport`) isn't associated with any one project--so I can't actually use `DataflowBlockFactory.CreateActionBlock`.

I'll update the code to include a comment explaining why we are using the banned API here.